### ISSUE TITLE
关于手写MOE模型的loss

### ISSUE THREAD

**Issue by jocskywalker** [bbruceyuan/LLMs-Zero-to-Hero#4]
你好大佬，这个负载均衡的模块也是Deepseek一个很重要的知识点，这里的loss计算是不是有点问题，这个expected_loss没有用上，另外多层的router——logits也可以加到代码里吗，谢谢大佬
![Image](https://github.com/user-attachments/assets/f0aa721d-520a-4c1b-9867-925c05012a61)

**Comment by bbruceyuan** [comment]
那个不是 loss，是预期的平衡 load 状态，因为 switchtransformers 那个 paper 里面用的 loss 就不需要这个东西，所以没有也无所谓，你可以和 GPT 聊聊，为什么现在这个 loss 计算是对的，以前有一个网友也质疑过这个点，但是最后发现不同的 api 计算出来的 loss 是一样的。
